### DISCUSSION TITLE
Using sqlalchemy_utils ChoiceType not working

### DISCUSSION THREAD

**Discussion by im-ajaymeena** [aminalaee/sqladmin#562] [Q&A] [answered]
Am I doing something wrong, I cant bring ChoiecType on admin page

**model file**

    class COLLECTION_TYPES(enum.Enum):
        1 = '1'
   
    class Collection(Base):
       id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
       name = Column(String, unique=True)
       user_id = Column("user", ForeignKey('user.id'))
       selected = Column(Boolean, default=False)

       created_at = Column(DateTime, default=datetime.datetime.now())
       updated_at = Column(DateTime, server_default=func.now(), onupdate=func.current_timestamp())
       user = relationship('User', backref='collection')
       num_entities = Column(Integer, default=0)

    
       collection_type = ChoiceType(COLLECTION_TYPES)
       encoded_meta_data = Column(String)

       @hybrid_property
       def milvus_collection_name(self) -> str:
           return f"_{self.id.replace('-', '_')}"

       __table_args__ = (UniqueConstraint(user_id, name),)

    ## alembic migration file (trimmed)
    
    def upgrade() -> None:
        op.add_column('collection', sa.Column('collection_type', sqlalchemy_utils.types.choice.ChoiceType(COLLECTION_TYPES), nullable=True))

    def downgrade() -> None:
        op.drop_column('collection', 'collection_type')
    
    
![image](https://github.com/aminalaee/sqladmin/assets/19550841/b6d7e522-dbec-45dd-933f-32dea07e47a5)

**Comment by aminalaee** [comment]
Hey @im-ajaymeena , what version of sqladmin are you using?

**Reply by im-ajaymeena** [reply]
@aminalaee  
My version is the latest one for sqladmin

sqladmin                      0.13.0
SQLAlchemy               2.0.19
SQLAlchemy-Utils       0.41.1

**Reply by aminalaee** [reply]
Yeah I think something wrong with your model definition: https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#module-sqlalchemy_utils.types.choice
Why isn't the `COLLECTION_TYPES` used on the model?

**Reply by im-ajaymeena** [reply]
Actually its inside the model, sorry for incorrect indentation, I have updated the question
 this is my actual code
 
![image](https://github.com/aminalaee/sqladmin/assets/19550841/56f9413b-c0b3-463d-aceb-566d7f9cdd92)

**Reply by aminalaee** [reply, marked as answer]
I'm not much familiar with sqlalchemy_utils but why do you have this?

```py
collection_type = ChoiceType(COLLECTION_TYPES)
```

But instead you should have:

```py
collection_type = Column(ChoiceType(COLLECTION_TYPES))
```

**Reply by im-ajaymeena** [reply]
Oh My bad, Thanks a lot :pray:
